### PR TITLE
Adding utils to get best order by lowest sender or highest signer amount

### PR DESCRIPTION
**Pull request by dmosites**
Get the best order by lowest sender or highest signer amount. E.g.:

```
const orders: Array<SignedOrder> = [...]
const best: SignedOrder = getByLowestSenderAmount(orders)
```

### CODE REVIEW

File: tools/utils/src/orders.ts [hunk 1]
```diff
@@ -95,3 +96,27 @@ export async function signOrder(
     },
   }
 }
+
+export function getByLowestSenderAmount(
```
Comment (poiresel) [inline tools/utils/src/orders.ts]: function name change to sortBy or getOrderBy?

Comment (dmosites) [reply tools/utils/src/orders.ts]: Happy to change to e.g. `getOrderByLowestSenderAmount` for consistency and clarity.

File: tools/utils/test/orders.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+            wallet: '',
+            kind: '',
+            token: '',
+            amount: String(i + highestAmount),
+          },
+        })
+      )
+    }
+    const best = getByHighestSignerAmount(orders)
+    expect(best.sender.amount).to.equal(String(highestAmount))
```
Comment (poiresel) [inline tools/utils/test/orders.ts]: should this be 55? not matching on 50

File: tools/utils/test/orders.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            wallet: '',
+            kind: '',
+            token: '',
+            amount: String(i + highestAmount),
+          },
+        })
+      )
+    }
+    const best = getOrderByHighestSignerAmount(orders)
+    expect(best.sender.amount).to.equal(String(highestAmount))
```
Comment (poiresel) [inline tools/utils/test/orders.ts]: for this test, are all the signer.Amount equivalent and just the sender Amount is changing?

File: tools/utils/test/orders.ts [hunk 4]
```diff
@@ -29,21 +29,21 @@ describe('Orders', async () => {
 
   it('Checks best order by highest signer', async () => {
     const orders = []
-    const count = 5
-    const highestAmount = 50
-    for (let i = 0; i < count; i++) {
+    const highestAmount = 5
+    let count = -1
```
Comment (poiresel) [inline tools/utils/test/orders.ts]: this should be 0, since the ++count you want it to start at 0 not -1.

**Pull request merged**